pkg/provider/aws/knowledgebase/cloudfront.go

```diff
@@ -0,0 +1,156 @@
+package knowledgebase
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/klothoplatform/klotho/pkg/collectionutil"
+	"github.com/klothoplatform/klotho/pkg/core"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base"
+	"github.com/klothoplatform/klotho/pkg/multierr"
+	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
+	"github.com/pkg/errors"
+)
+
+var CloudfrontKB = knowledgebase.Build(
+	knowledgebase.EdgeBuilder[*resources.CloudfrontDistribution, *resources.S3Bucket]{
+		Expand: func(distro *resources.CloudfrontDistribution, bucket *resources.S3Bucket, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			var errs multierr.Error
+			for consRef := range distro.ConstructsRef {
+				conn := s3ToCloudfrontConnection{
+					distro:    distro,
+					bucket:    bucket,
+					dag:       dag,
+					construct: consRef,
+				}
+				oai, err := conn.createOai()
+				if err != nil {
+					errs.Append(err)
+					continue
+				}
+				err = conn.attachPolicy(oai)
+				errs.Append(err)
+			}
+			return errs.ErrOrNil()
+		},
+		Configure: func(distro *resources.CloudfrontDistribution, bucket *resources.S3Bucket, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			distro.DefaultRootObject = bucket.IndexDocument
+			return nil
+		},
+	},
+	knowledgebase.EdgeBuilder[*resources.CloudfrontDistribution, *resources.ApiStage]{
+		Expand: func(distro *resources.CloudfrontDistribution, stage *resources.ApiStage, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			var gwId string
+			switch len(stage.ConstructsRef) {
+			case 0:
+				return errors.Errorf(`couldn't determine the id of the construct that created API stage "%s"`, stage.Id())
+			case 1:
+				cons, _ := collectionutil.GetOneEntry(stage.ConstructsRef)
+				gwId = cons.ID
+			default:
+				var ids []string
+				for _, cons := range collectionutil.Keys(stage.ConstructsRef) {
+					ids = append(ids, cons.ID)
+				}
+				sort.Strings(ids)
+				return errors.Errorf(
+					`couldn't determine the id of the construct that created API stage "%s": expected just one construct, but found [%s]`,
+					stage.Id(),
+					strings.Join(ids, ", "))
```

Comment:
```suggestion
				return errors.Errorf(
					`couldn't determine the id of the construct that created API stage "%s": expected just one construct, but found %v`,
					stage.Id(),
					ids)
```
nit: %v will do the joining and bracketing for you for arrays